bag_of_words/code/bag_of_words.py

```diff
@@ -0,0 +1,36 @@
+# Read papers and convert them to bag-of-words representations.
+
+indir = "../data/raw/pubmed-txts-mf/"
+outdir = "../data/processed/"
+from sklearn.feature_extraction.text import CountVectorizer
+vectorizer = CountVectorizer()
+import os
+import numpy as np
+import pickle
+
+# Read papers into a list
+pmids = os.listdir(indir)
+pmids.sort()
+# pmids = pmids[:5]  # For debugging
+corpus = [''] * len(pmids)
+print("Reading corpus ...")
+for i in range(len(pmids)):
+    if i % 10000 == 0:
+        print(str(i) + " docs processed ...")
+    for field in os.listdir(indir + pmids[i] + '/'):
+        with open(indir + pmids[i] + '/' + field, 'r') as fin:
+            text = fin.read().replace('\n', ' ')
+            corpus[i] += text
+print('')
+
+# Convert papers to bags of words, and save objects to file
+print("Building bag-of-words representation ...")
+bow = vectorizer.fit_transform(corpus)

```

Comment:
If this were a `def vectorize_corpus` function, it could default to `CountVectorizer` to make the `bow`, but it could also accept a vectorizer object as an argument, in which case, you could simply call the `bow` variable something like `vectors`, which would generalize to TF-IDF or other tokenizers and vectorizers.